docs/rfcs/1005-realm-key-rotation.md

```diff
@@ -0,0 +1,924 @@
+removing realm vlobs (cannot remove a realm given it is related to certificates) or
+organization form the database.
+
+### 5.3 - Server data migration
+
+Drop tables:
+
+- `vlob_encryption_revision`
+- `message`
+
+Table `vlob_atom`:
+
+- Remove all rows that contains old encryption revision data (i.e. the table only contains
+  the vlob currently accessible by the users).
+- Remove column `vlob_encryption_revision`
+- Add column `key_index` (`INTEGER NOT NULL`). This defaults to 1 when migrating data.
+
+Table `realm`
+
+- Remove columns `encryption_revision`, `maintenance_started_by`, `maintenance_started_on` and `maintenance_type`
+
+## 6 - Changes: Client-side
+
+### 6.1 - Certificate Ops
+
+`RealmKeyRotationCertificate` should be checked (only OWNER of the realm can create such
+certificate).
+
+`RealmNameCertificate` should be checked (only OWNER of the realm can create such
+certificate).
+
+### 6.1 - Workspace name synchronization & default value
+
+There is no atomicity between realm creation an the upload of the first `RealmNameCertificate`.
+
+Hence a client may end up with a realm without realm name certificate (and in such case the
+name of the realm defaults to the realm ID).
+
+So that any realm OWNER detecting the `RealmNameCertificate` is missing should look in
+it local user manifest if a workspace name is specified, and do a `realm_rename` in
+that case.
+
+This approach has the benefit of transparently handling the legacy (i.e. realm created,
+but no realm name certificate).
+
+### 6.2 - Workspaces listing
+
+The client has to rely on both the certificates and the local user manifest to get
+the list of workspaces and their corresponding name:
+
+1) Get all the realm for which the user has a realm role certificate with currently
+   active role.
+2) Get the name from the `RealmNameCertificate` if available and valid.
+3) Otherwise, look in the local user manifest for a workspace name entry.
+4) Otherwise default to the realm ID
+
+### 6.3 - Fetching keys-bundle
+
+The client lazily fetch the realm keys-bundle: it does the fetch operation when a server
+operation requires it (i.e. `realm_share`, `vlob_read`, `vlob_create` and `vlob_update`).
+
+The realm keys-bundle is only kept in RAM, not on disk.
+
+Once fetched, keys-bundle must be checked against its corresponding `RealmKeyRotationCertificate`
+for same author/timestamp/realm_id/key_index.
+
+Each key in the Keys-bundle must be checked against the canary in `RealmKeyRotationCertificate`
+that introduced it.
+
+At any time, `realm_share`/`vlob_read`/`vlob_create` can return `bad_key_index` if the client
+consider the wrong key to be the last one. In this case the client simply forgets its realm
+keys-bundle and re-do the fetch operation.
+
+### 6.4 - Keys-bundle self-healing
+
+A keys-bundle can be corrupted in multiple ways:
+
+1. Invalid data: bad signature, bad encryption, bad serialization
+2. Additional keys
+3. Missing keys
```

Comment:
```suggestion
- Invalid data: bad signature, bad encryption, bad serialization
- Additional keys
- Missing keys
```

I don't think that list need to be ordered

Reply:
you're right (I used the number to reference the items in the next paragraph but that's no longer the case)